### PR TITLE
Automatically apply fixes to hosts file

### PR DESCRIPTION
**Pull request by yashgorana**
## Description

Relates to https://github.com/OpenMined/Heartbeat/issues/780

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -927,52 +927,79 @@ commands =
...
-    bash -c 'k3d registry create registry.localhost --port 12345 -v $HOME/.k3d-registry:/var/lib/registry'
-    bash -c '\
-        CYAN="\033[1;36m"; \
-        CLEAR="\033[0m"; \
-        printf "\n${CYAN}---------------------------------------------------------\n" ;\
-        printf "Make sure to add the following to your /etc/hosts file:${CLEAR}\n";\
-        printf "127.0.0.1 k3d-registry.localhost\n";\
-        printf "${CYAN}---------------------------------------------------------${CLEAR}\n"'
+    bash -c 'k3d registry create registry.localhost --port 12345 -v $HOME/.k3d-registry:/var/lib/registry || true'
+    sudo {envpython} scripts/patch_hosts.py --add-k3d-registry --fix-docker-hosts
```
Comment (madhavajay) [inline tox.ini:953]: Will `sudo` work from powershell / cmd.exe on windows?

Comment (yashgorana) [reply tox.ini:953]: Not as of now - I'll have a fix in a separate PR

File: scripts/patch_hosts.py [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+        elif Platform.linux():
+            p = "/etc/hosts"
+        elif Platform.macos():
+            new_hosts = "/etc/hosts"
+            old_hosts = "/private/etc/hosts"
+            p = new_hosts if os.path.exists(new_hosts) else old_hosts
+        elif Platform.windows():
+            p = "C:\Windows\System32\drivers\etc\hosts"
+        else:
+            raise Exception(f"Unsupported OS: {Platform.system()}")
```
Comment (rasswanth-s) [inline scripts/patch_hosts.py]: NIT: Platform.system is a string, we could remove the brackets

Comment (yashgorana) [reply scripts/patch_hosts.py]: eesh this would've caused an exception. pushed a fix.

File: scripts/patch_hosts.py [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+    if not Platform.windows():
+        return os.geteuid() == 0
+    else:
+        # stdlib
+        import ctypes
+
+        return ctypes.windll.shell32.IsUserAnAdmin() == 1
+
+
+def wsl2_disable_auto_hosts() -> None:
```
Comment (rasswanth-s) [inline scripts/patch_hosts.py:103]: Q: Yash , could you remind me again, on why we remove this?

File: tox.ini [hunk 4]
```diff
@@ -937,52 +943,79 @@ commands =
...
 
 [testenv:dev.k8s.start]
-changedir = {toxinidir}/packages/grid
+description = Start local Kubernetes registry & cluster with k3d
+changedir = {toxinidir}
+passenv=*
```
Comment (rasswanth-s) [inline tox.ini:958]: Q: Why do we pass in all the environment variables?

File: tox.ini [hunk 5]
```diff
@@ -937,52 +943,79 @@ commands =
...
-    bash -c '\
-        devspace use context k3d-syft-dev; \
-        devspace use namespace syft; \
-        devspace deploy --var CONTAINER_REGISTRY=k3d-registry.localhost:12345 --config=devspace.yaml'
+    bash -c 'devspace deploy -b --kube-context k3d-syft-dev --namespace syft --var CONTAINER_REGISTRY=k3d-registry.localhost:12345'
+
+[testenv:dev.k8s.info]
+description = Gather info about the localKubernetes cluster
+passenv=HOME, USER
+ignore_errors = True
```
Comment (rasswanth-s) [inline tox.ini:983]: Q: Why do we ignore errors at this place?

File: tox.ini [hunk 6]
```diff
@@ -937,52 +943,79 @@ commands =
...
         rm -rf .devspace; echo ""; \
         k3d cluster delete syft-dev; echo ""; \
-        k3d registry delete registry.localhost; echo "";\
         kubectl config view'
+
+[testenv:dev.k8s.destroyall]
+description = Destroy both local Kubernetes cluster and registry
+changedir = {toxinidir}
+passenv=HOME, USER
+ignore_errors=True
```
Comment (rasswanth-s) [inline tox.ini:1011]: Q: Why do we ignore error at this place?

File: tox.ini (hunk 1 above)
Comment (rasswanth-s) [reply tox.ini:953]: Logged an Issue for future: https://github.com/OpenMined/Heartbeat/issues/833

**Review by rasswanth-s — APPROVED** [review]
Great work @yashgorana , left only some questions

**Pull request merged**